client/src/main/java/com/vesoft/nebula/client/meta/MetaManager.java

```diff
@@ -0,0 +1,498 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+package com.vesoft.nebula.client.meta;
+
+import com.facebook.thrift.TException;
+import com.google.common.collect.Maps;
+import com.google.common.net.HostAndPort;
+import com.vesoft.nebula.client.meta.exception.ExecuteFailedException;
+import com.vesoft.nebula.meta.EdgeItem;
+import com.vesoft.nebula.meta.IdName;
+import com.vesoft.nebula.meta.TagItem;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * MetaManager is a manager for meta info, such as spaces,tags and edges.
+ * MetaManager is composed of a cache named {@link MetaInfo} and real metaClient.
+ * How to use:
+ * MetaManager manager = MetaManager.getMetaManager(Arrays.asList(HostAndPort.fromString(addr)));
+ * manager.init();
+ */
+public class MetaManager {
+    private static final Logger LOGGER = LoggerFactory.getLogger(MetaManager.class);
+
+    private static MetaClient metaClient;
+    private static MetaManager metaManager;
+    private final ReentrantReadWriteLock lock = new ReentrantReadWriteLock();
```

Comment:
why not move the lock to MetaInfo, the MetaInfo make itself thread-safe. and the MetaManager only call the `get` or `set` function by MetaInfo

Reply:
no need,  the set functions are just put operators.